{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting started\n",
    "\n",
    "Follow this gide to install the basic software and data requiered by this project"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Notebook\n",
    "\n",
    "This is a Jupyter Notebook to show how to install all software required by this project\n",
    "\n",
    "### Requirements\n",
    "#### Basic programs\n",
    "1. Python3\n",
    "2. virtualenv3\n",
    "3. C++\n",
    "\n",
    "#### Bioinformatics programs\n",
    "These programs have to be installed and included in the user PATH variable.\n",
    "\n",
    "* STAR (https://github.com/alexdobin/STAR)\n",
    "* SAMTools (http://www.htslib.org/)\n",
    "* BAMTools (https://github.com/pezmaster31/bamtools)\n",
    "* TPMCalculator (https://github.com/r78v10a07/TPMCalculator)\n",
    "\n",
    "### Instalation of the Notebook\n",
    "\n",
    "Execute this commands in a BASH terminal just the first time to get the notebook dependencies installed.\n",
    "\n",
    "```\n",
    "# Creating a virtual environment\n",
    "$ virtualenv3 venv\n",
    "$ source venv/bin/activate\n",
    "$ pip install -r requirements/jupyter.txt\n",
    "$ pwd\n",
    "```\n",
    "\n",
    "Edit the file in config/proj_env and change the PROJ_ROOT variable (first line)\n",
    "\n",
    "### Downloading genomes files\n",
    "\n",
    "* hg38 ftp://igenome:G3nom3s4u@ussd-ftp.illumina.com/Homo_sapiens/UCSC/hg38/Homo_sapiens_UCSC_hg38.tar.gz \n",
    "* hg19 ftp://igenome:G3nom3s4u@ussd-ftp.illumina.com/Homo_sapiens/UCSC/hg19/Homo_sapiens_UCSC_hg19.tar.gz\n",
    "\n",
    "Uncompress this file and creates a data folder.\n",
    "\n",
    "```\n",
    "$ mkdir data\n",
    "$ cd data\n",
    "$ wget ftp://igenome:G3nom3s4u@ussd-ftp.illumina.com/Homo_sapiens/UCSC/hg38/Homo_sapiens_UCSC_hg38.tar.gz \n",
    "$ wget ftp://igenome:G3nom3s4u@ussd-ftp.illumina.com/Homo_sapiens/UCSC/hg19/Homo_sapiens_UCSC_hg19.tar.gz\n",
    "$ tar xzfv Homo_sapiens_UCSC_hg38.tar.gz\n",
    "$ tar xzfv Homo_sapiens_UCSC_hg19.tar.gz\n",
    "$ mkdir hg19 hg38\n",
    "$ mv Homo_sapiens/UCSC/hg38/Annotation/Genes/genes.gtf hg38/\n",
    "$ mv Homo_sapiens/UCSC/hg38/Sequence/WholeGenomeFasta/genome.fa hg38/\n",
    "$ mv Homo_sapiens/UCSC/hg19/Annotation/Genes/genes.gtf hg19/\n",
    "$ mv Homo_sapiens/UCSC/hg19/Sequence/WholeGenomeFasta/genome.fa hg19/\n",
    "\n",
    "```\n",
    "\n",
    "### Instalation of software \n",
    "\n",
    "#### Compiling the STAR\n",
    "\n",
    "```\n",
    "$ cd src/\n",
    "$ git clone https://github.com/alexdobin/STAR\n",
    "$ cd STAR/source\n",
    "$ make STAR\n",
    "$ cd ../../../bin/\n",
    "$ ln -s ../src/STAR/source/STAR STAR\n",
    "$ cd ../\n",
    "$ cd data/hg19/\n",
    "$ mkdir STAR\n",
    "$ cd STAR\n",
    "$ ln -s ../genome.fa genome.fa\n",
    "$ ln -s ../genes.gtf genes.gtf\n",
    "$ ../../../bin/STAR --runThreadN 16 --runMode genomeGenerate --genomeDir . --genomeFastaFiles genome.fa --sjdbGTFfile ./genes.gtf\n",
    "$ cd ../../../\n",
    "```\n",
    "\n",
    "#### Compiling the BAMTools\n",
    "\n",
    "```\n",
    "$ cd src/\n",
    "$ git clone https://github.com/pezmaster31/bamtools\n",
    "$ cd bamtools\n",
    "$ mkdir build\n",
    "$ cd build\n",
    "$ cmake ..\n",
    "$ make\n",
    "$ cd ../../../bin/\n",
    "$ ln -s ../src/bamtools/bin/bamtools bamtools\n",
    "$ cd ../\n",
    "$ source config/proj_env\n",
    "```\n",
    "\n",
    "#### Compiling TPMCalculator\n",
    "\n",
    "```\n",
    "$ cd src/\n",
    "$ git clone https://github.com/r78v10a07/TPMCalculator\n",
    "$ cd TPMCalculator\n",
    "$ make \n",
    "$ cd ../../bin/\n",
    "$ ln -s ../src/TPMCalculator/bin/TPMCalculator TPMCalculator\n",
    "$ cd ../\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.4.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
